src/b_spline.cc

```diff
@@ -0,0 +1,96 @@
+/* Copyright 2018 Chair for Computational Analysis of Technical Systems, RWTH Aachen University
+
+This file is part of SplineLib.
+
+SplineLib is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation version 3 of the License.
+
+SplineLib is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along with SplineLib.  If not, see
+<http://www.gnu.org/licenses/>.
+*/
+
+#include "b_spline.h"
+
+#include <algorithm>
+#include <functional>
+#include <numeric>
+
+#include "basis_function_factory.h"
+
+BSpline::BSpline(KnotVector knot_vector, Degree degree, std::vector<ControlPoint> control_points) :
```

Comment:
Please pass `knot_vector` and `control_points` as const reference. This should increase performance.